frontend/declarations.d.ts

```diff
@@ -1,4 +1,6 @@
-import { AppThemeType } from "./src/app/providers/ThemeProvider";
+import type { AppThemeType } from "./src/app/providers/ThemeProvider";
+
+export type { AppState } from "./src/app/providers/ReduxProvider/store";
```

Comment:
export type AppState is not correct from here, because declarations file only need to declare something globally and use types without imports from this file. So, I would suggest to move store.ts to shared folder because we use there reducers from shared folder as well and we need AppState to be available across whole app. 
Create folder shared/providers/ReduxProvider and put all content here from app/providers/ReduxProvider, and do not forget to fix import issues